docs/development.rst

```diff
@@ -254,28 +254,140 @@ HMC inventory file
+code that uses the :mod:`zhmcclient.testutils` module, such as the end2end tests,
+the `example scripts`_, other zhmcclient projects, or even projects by users of
+the zhmcclient library.
 
-.. _examples directory: https://github.com/zhmcclient/python-zhmcclient/tree/master/examples
+.. _example scripts: https://github.com/zhmcclient/python-zhmcclient/tree/master/examples
 
-Its format is described in the comment header of the
-`example HMC inventory file <https://github.com/zhmcclient/python-zhmcclient/blob/master/examples/example_hmc_inventory.yaml>`_.
+The HMCs and HMC groups defined in the HMC inventory file have nicknames and
+the nickname to be used for end2end tests and the example scripts can be
+specified using the ``TESTHMC`` environment variable, for example:
 
-The format is compatible to the format of Ansible inventory files in YAML format,
-with some extensions and some limitations:
+.. code-block:: bash
 
-Extensions:
+    $ TESTHMC=HMC1 make end2end             # run end2end tests against nickname "HMC1"
 
-* The HMC inventory files define and document specific variables for the
-  HMC hosts.
+    $ TESTHMC=HMC1 examples/list_cpcs.py    # run this example script against nickname "HMC1"
 
-Limitations:
+If no nickname is specified using the ``TESTHMC`` environment variable, the
+nickname "default" is used, for example:
+
+.. code-block:: bash
+
+    $ make end2end             # run end2end tests against nickname "default""
+
+    $ examples/list_cpcs.py    # run this example script against nickname "default""
+
+By default, the HMC inventory file ``~/.zhmc_inventory.yaml`` is used.
+A different path name can be specified with the ``TESTINVENTORY`` environment
+variable.
+
+The following describes the structure of the HMC inventory file:
+
+.. code-block:: yaml
+
+    all:                                # Nickname of the top-level HMC group; must be 'all'.
+
+      hosts:                            # Definition of all HMCs, each with an item as follows:
+
+        <hmc_nick>:                     # Nickname of an HMC; may be a DNS hostname, IP address,
+                                        #   or arbitrary string.
+
+          description: <string>         # Optional: One line description of the HMC.
+
+          contact: <string>             # Optional: Informal reference to a contact for the HMC.
+
+          access_via: <string>          # Optional: Reminder on network setup needed for access.
+
+          ansible_host: <host>          # If real HMC: DNS hostname or IP address of HMC, if
+                                        #   arbitrary string was used as HMC nickname.
+
+          mock_file: <path_name>        # If mocked HMC: Relative path name of HMC mock file.
+          cpcs:                         # CPCs to test against. Can be a subset or all CPCs
+                                        #   managed by the HMC.
+
+            <cpc_name>:                 # CPC name.
```

Comment:
minor: it would be great if we could somehow reflect our internal test setups here.
As in: it would be nice if I could say: "just pick the one SE you will find on the HMC I specified".

Reply:
I have created issue https://github.com/zhmcclient/python-zhmcclient/issues/1134 for this idea.